### ISSUE TITLE
Agent invalid RSA siguature

### ISSUE THREAD

**Issue by trenton92** [Ylianst/MeshCentral#1681]
Hello
How to determine which agents have invalid RSA?
Thanks
![image](https://user-images.githubusercontent.com/61736438/89390622-a58a3180-d70f-11ea-887b-c8d775af6f00.png)

**Comment by vitko-bg** [comment]
And also "bad signatures" ???
![image](https://user-images.githubusercontent.com/59362521/95065442-f8c13600-06f8-11eb-91bc-2074534fc310.png)

All my Linux and Mac agent are not appearing on my MC Server, looks like all of them are with Bad Signatures....
How I can check this also.

**Comment by githubbiswb** [comment]
Same boat here and I also have Linux agents so I have a hunch that might be the place where it is, although all of my Linux agents are all connected, so I am not having that issue

**Comment by vitko-bg** [comment]
Is your MC server behind proxy? My one is. Can this be a NGINX proxy issue?

**Comment by githubbiswb** [comment]
> 
> 
> Is your MC server behind proxy? My one is. Can this be a NGINX proxy issue?

Yep, it is behind a proxy but what makes my case a little different is I only have 5 that are having an issue (at one time I had 8, but then I deleted 3 Linux hosts, didn't connect those dots until your post). 

However, I have 19 total Linux hosts that I have a mesh agent installed on so that means 14 are working just fine or aren't reporting they have an issue, if this is indeed a Linux -> Nginx Reverse Proxy -> Meshcentral issue going here. And all 19 go through the reverse proxy, because I have everything in a Docker Swarm, so by definition all of my Linux hosts are coming from the WAN side of Nginx, which then forwards it on over to Meshcentral. 

For what it is worth, I also don't use the LE features of MeshCentral, not sure if you do, but have a dedicated container that takes care of that, and I simply script the copy and rename of the certs into the correct folders for mesh to use.

**Comment by githubbiswb** [comment]
The plot thickens. 
I Exported all of my devices as a csv, 114 in total connected
I put in the line for ignoring agent hashes in my config.json. 
          "ignoreagenthashcheck" : true,
Brought my meshcentral container down
Brought it back up
Got the warnings in the logs that I shouldn't run with hash disabled
Brought everything back up
Exported my devices as a csv, still 114 connected.
I commented out the line for ignoring agent hashes in my config.json. 
          "_ignoreagenthashcheck" : true,
Brought mesh down
Brought it back up
Now only 1 agent complains about being incorrect
Still 114 devices connected. 

So it seems my agent hash complaints aren't keeping my agents from connecting, or when I brought it down and said to ignore the one(s) that was/were having the problem still didn't check in.

**Comment by PetieM** [comment]
Did anyone ever figure out how to determine which agent or agents are contributing to that count of invalid/bad signatures? I'm seeing one on my server now but can't seem to find it.

**Comment by PetieM** [comment]
For anyone else who finds this wondering the same thing, you can determine which agent or agents are causing these counters to increase by going to the server console and entering "agentissues" - it will list all current agent issues detected by the server along with the IP address of the agent in question. You should then be able to easily match that to an agent either via the IP address column in list view or by using the filter.

**Comment by black3dynamite** [comment]
> For anyone else who finds this wondering the same thing, you can determine which agent or agents are causing these counters to increase by going to the server console and entering "agentissues" - it will list all current agent issues detected by the server along with the IP address of the agent in question. You should then be able to easily match that to an agent either via the IP address column in list view or by using the filter.

Entering "dupagents" will list the duplicate agents
